src/objects/basic_npc.py

```diff
@@ -83,3 +98,30 @@ def collide(self, obstacle_pos):
 
     def accept(self, visitor):
         return visitor.visit_basic_npc(self)
+    
+    def update_conversation(self, player_pos, keys):
+        if not self.conversation_finished and not self.is_in_conversation:
+            distance = self.pos.distance_to(player_pos)
+            if distance < 100:  # Trigger conversation if player is close enough
+                self.start_conversation()
```

Comment:
This has a lot of issues so far, it starts against the player's will and different conversations overlap. I think I will try to tweak the behaviour a bit before merging.